tests/test_lightcurve_fitting.py

```diff
@@ -1,483 +1,467 @@
-        # Test to pass
-        npts = 1234
-        time, _, _, _ = simulations.simulate_lightcurve('WASP-107b', 0.1,
-                                                        npts=npts, plot=False)
-        self.assertEqual(len(time), npts)
-
-        # Test to fail
-        self.assertRaises(KeyError, simulations.simulate_lightcurve,
-                          'foobar', 0.1)
+
+class testingMetaClass(S5MetaClass):
+    """Apply testing-relevant defaults to S5MetaClass objects.
+
+    Parameters
+    ----------
+    meta : eureka.S5_lightcurve_fitting.s5_meta.S5MetaClass
+        The meta object to prepare.
+    """
+    def __init__(self, folder=None, file=None, eventlabel=None, **kwargs):
+        super().__init__(folder, file, eventlabel, **kwargs)
+        self.spec_hw = 1
+        self.bg_hw = 2
+        self.fit_par = ''
+        self.fit_method = 'lsq'
+        self.run_myfuncs = []
+        self.topdir = './'
+        self.inputdir = ''
+        self.outputdir = ''
+        self.sharedp = False
+        self.whitep = False
+        self.set_defaults()
+
+# Tests for the parameters.py module
+
+
+def test_parameter(capsys):
+    """Test that a Parameter object can be created"""
+    # Create the parameter
+    pname = 'p1'
+    pval = 12.34
+    ptype = 'free'
+    priorpar1 = 10
+    priorpar2 = 15
+    prior = 'U'
+    param = Parameter(pname, pval, ptype, priorpar1, priorpar2, prior)
+
+    # Test bogus input
+    with pytest.raises(TypeError):
+        Parameter(123)
+    with pytest.raises(ValueError):
+        Parameter('foo', 123, 123)
+
+    # Test the attributes
+    assert param.name == pname
+    assert param.value == pval
+    assert param.ptype == ptype
+    assert param.priorpar1 == priorpar1
+    assert param.priorpar2 == priorpar2
+    assert param.prior == prior
+    assert param.values == [pname, pval, ptype, priorpar1, priorpar2, prior]
+
+
+def test_parameters(capsys):
+    """Test that a Parameters object can be created"""
+    params = Parameters()
+    params.param1 = 123.456, 'free'
+    params.param2 = 234.567, 'free', 200, 300
+
+    # Test the auto attribute assignment
+    assert params.param1.values == ['param1', 123.456, 'free']
+    assert params.param2.values == ['param2', 234.567, 'free', 200, 300]
+    # Test FileNotFoundError
+    with pytest.raises(FileNotFoundError):
+        Parameters(None, 'non-existent.epf')
+
+# Tests for the models.py module
+
+
+def test_model(capsys):
+    """Tests for the generic Model class"""
+    # Test model creation
+    name = 'Model 1'
+    model = models.Model(name=name)
+    assert model.name == name
+
+    # Test model units
+    assert str(model.time_units) == 'BMJD_TDB'
+    model.time_units = 'MJD'
+    assert model.time_units == 'MJD'
+
+
+def test_compositemodel(capsys):
+    """Tests for the CompositeModel class"""
+    model1 = models.Model()
+    model2 = models.Model()
+    comp_model = model1*model2
+    comp_model.name = 'composite'
+
```

Comment:
Do we need an assertion for this test?

Reply:
I don't think so, no. This just tests that two models can be combined using multiplication (using the `Model.__mul__` function) which is a utility we've never used.